From 3470a2f9376958478886ad90f85df6e97a5c0f07 Mon Sep 17 00:00:00 2001
From: Hannes Frederic Sowa <hsowa@redhat.com>
Date: Tue, 3 Jun 2014 13:23:46 -0400
Subject: [lib] random32: add __init prefix to prandom_start_seed_timer

Message-id: <1401801829-9802-8-git-send-email-hsowa@redhat.com>
Patchwork-id: 82390
O-Subject: [RHEL7.1 PATCH net 07/10] random32: add __init prefix to prandom_start_seed_timer
Bugzilla: 1100299
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Florian Westphal <fwestpha@redhat.com>
RH-Acked-by: Daniel Borkmann <dborkman@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1100299
Upstream Status: 66b251422be7cb39e7619fee647724720f74d1f7
Brew: http://brewweb.devel.redhat.com/brew/taskinfo?taskID=7530240
Tested: verified with prandom kernel test suite

Upstream Commit:
commit 66b251422be7cb39e7619fee647724720f74d1f7
Author: Daniel Borkmann <dborkman@redhat.com>
Date:   Tue Nov 12 23:45:41 2013 +0100

    random32: add __init prefix to prandom_start_seed_timer

    We only call that in functions annotated with __init, so add __init
    prefix in prandom_start_seed_timer() as well, so that the kernel can
    make use of this hint and we can possibly free up resources after it's
    usage. And since it's an internal function rename it to
    __prandom_start_seed_timer().

    Signed-off-by: Daniel Borkmann <dborkman@redhat.com>
    Signed-off-by: Hannes Frederic Sowa <hannes@stressinduktion.org>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Hannes Frederic Sowa <hsowa@redhat.com>
Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/lib/random32.c b/lib/random32.c
index 82da4f4..4f9d5df 100644
--- a/lib/random32.c
+++ b/lib/random32.c
@@ -222,7 +222,7 @@ static void __prandom_timer(unsigned long dontcare)
  add_timer(&seed_timer);
 }
 
-static void prandom_start_seed_timer(void)
+static void __init __prandom_start_seed_timer(void)
 {
  set_timer_slack(&seed_timer, HZ);
  seed_timer.expires = jiffies + 40 * HZ;
@@ -270,7 +270,7 @@ void prandom_reseed_late(void)
 static int __init prandom_reseed(void)
 {
  __prandom_reseed(false);
- prandom_start_seed_timer();
+ __prandom_start_seed_timer();
  return 0;
 }
 late_initcall(prandom_reseed);
-- 
1.7.1